user/how-to-guides/backup-emergency-restore-v4.md

```diff
@@ -66,44 +66,33 @@ any GNU/Linux system.
 
         [user@restore ~]$ sudo dnf install rpmdevtools
 
- 5. Extract the `scrypt` binary from the RPM.
+ 5. Extract the `scrypt` binary from the RPM and make it conveniently
+    available.
 
         [user@restore ~]$ rpmdev-extract scrypt-*.rpm
+        [user@restore ~]$ alias scrypt="$PWD/scrypt-*/usr/bin/scrypt"
```

Comment:
This doesn't work for me in a Debian 11 minimal qube:

```
$ ls scrypt-1.2.1-2 
scrypt-1.2.1-2
$ alias scrypt="$PWD/scrypt-*/usr/bin/scrypt"
[follow instructions]
$ echo "backup-header!$backup_pass" |\
    scrypt dec -P backup-header.hmac backup-header.verified && \
    diff -qs backup-header backup-header.verified
bash: /home/user/scrypt-*/usr/bin/scrypt: No such file or directory
$ alias
alias scrypt='/home/user/scrypt-*/usr/bin/scrypt'
```

Reply:
Simply doing something like `alias scrypt="$PWD/scrypt-1.2.1-2"` works fine, though.

Reply:
Oooh you probably got that `scrypt-1.2.1-2` binary directly from your backup, not by running `rpmdev-extract scrypt-*.rpm`?

Reply:
I already had that binary. I've had it for so long that I can't remember how I originally got it. It could have very well originally come from running `rpmdev-extract scrypt-*.rpm`, but I can't recall, as it was years ago.

Reply:
Makes sense. If you aren't running the `rpmdev-extract` step at the moment (which would have created a directory tree `scrypt-VERSION/usr/bin/` and put the executable named `scrypt` inside of it) you have to place the differently obtained executable into an equivalent directory tree or adjust the alias to refer to where it ended up.

Reply:
Since we explicitly recommend that users keep a copy of the `scrypt` binary with their backups, I think we should assume that at least some of them will follow that advice, and therefore we shouldn't assume that they're doing `rpmdev-extract` at the moment. (We should consider, for example, that they might be trying to follow this procedure completely offline, with no access to the internet, just a saved copy of these instructions, their backup, and whatever binaries they have saved or installed on their local system, which might just be some old Ubuntu disc.)

Reply:
That whole section might need a rewrite too, if it's to be reliable in both the "obtain necessary tools right now" and "(1) download them all in advance, (2) make them available when it's time" scenarios. E.g. that `scrypt` binary is dynamically linked so it depends on a compatible version of the OpenSSL libraries being installed in the rescue environment. (Hmm, maybe it should be statically linked)